ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productteaser/v1/productteaser/clientlibs/js/actions.js

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+'use strict';
+
+class ProductTeaser {
+    constructor(rootElement) {
+        rootElement.forEach(node => {
```

Comment:
I personally find this a bit counter intuitive to have one instance of `ProductTeaser` taking care of all the product teasers on a page (see https://github.com/adobe/aem-core-cif-components/blob/master/ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/clientlibs/js/product-carousel.js#L103-L108). On the other hand, you don't store any state in the class (instead it's in the DOM), so one instance of `ProductTeaser` per AEM component is not really needed.

Reply:
Actually, that's really stupid of me. I should've done like in the product carousel.